src/main/kotlin/com/markskelton/ThemeConstructor.kt

```diff
@@ -0,0 +1,240 @@
+import java.util.*
+import javax.swing.UIManager
+
+enum class ColorVariant {
+  VIVID, NORMAL
+}
+
+enum class FontVariant(val schemeValue: Int) {
+  BOLD(1), ITALIC(2), BOLD_ITALIC(3), NONE(0)
+}
+
+data class ColorPalette(
+  val variant: ColorVariant,
+  val colors: Map<String, String>
+)
+
+object ThemeConstructor {
+  private val gson = Gson()
+  private val logger = Logger.getInstance(this::class.java)
+  private const val ONE_DARK_FILE_PREFIX = "one-dark-"
+
+  fun constructNewTheme(newSettings: ThemeSettings): UIManager.LookAndFeelInfo =
+    constructLookAndFeel(getUpdatedEditorScheme(newSettings))
+
+  fun useExistingTheme(): UIManager.LookAndFeelInfo =
+    constructLookAndFeel(getPreExistingTheme())
+
+  private fun getPreExistingTheme(): Path =
+    findConstructedThemes(getAssetsDirectory())
+      .findFirst()
+      .orElseGet {
+        getUpdatedEditorScheme(ThemeSettings.instance)
+      }
+
+  private fun constructLookAndFeel(updatedEditorScheme: Path): TempUIThemeBasedLookAndFeelInfo {
+    val oneDarkLAF = LafManagerImpl.getInstance().installedLookAndFeels
+      .filterIsInstance<UIThemeBasedLookAndFeelInfo>()
+      .first {
+        it.theme.id == ONE_DARK_ID
+      }
+    return TempUIThemeBasedLookAndFeelInfo(
+      oneDarkLAF.theme,
+      VfsUtil.findFile(updatedEditorScheme, true)
+    )
+  }
+
+  private fun getUpdatedEditorScheme(themeSettings: ThemeSettings): Path {
+    val assetsDirectory = getAssetsDirectory()
+    cleanDirectory(assetsDirectory)
+    // Intellij caches files, and will not update if you change the contents of the file
+    val newEditorSchemeFile = Paths.get(
+      assetsDirectory.toAbsolutePath().toString(),
+      "$ONE_DARK_FILE_PREFIX${UUID.randomUUID()}.xml"
+    )
+    buildNewEditorScheme(themeSettings, newEditorSchemeFile)
+    return newEditorSchemeFile
+  }
+
+  private fun cleanDirectory(assetsDirectory: Path) {
+    findConstructedThemes(assetsDirectory)
+      .forEach { Files.delete(it) }
+  }
+
+  private fun findConstructedThemes(assetsDirectory: Path) = Files.walk(assetsDirectory)
+    .filter { it.fileName.toString().startsWith(ONE_DARK_FILE_PREFIX) }
+
+  private fun buildNewEditorScheme(themeSettings: ThemeSettings, newSchemeFile: Path) {
+    val colorPalette = getColorPalette(themeSettings)
+    val editorTemplate = getEditorXMLTemplate()
+    val updatedScheme = applySettingsToTemplate(
+      editorTemplate,
+      themeSettings,
+      colorPalette
+    )
+    writeXmlToFile(newSchemeFile, updatedScheme)
+  }
+
+  private fun applySettingsToTemplate(
+    editorTemplate: Node,
+    themeSettings: ThemeSettings,
+    colorPalette: ColorPalette
+  ): Node {
+    val (paletteVariant, colors) = colorPalette
+    val themeTemplate = editorTemplate.clone() as Node
+    themeTemplate.breadthFirst()
+      .map { it as Node }
+      .forEach {
+        when (it.name()) {
+          "scheme" -> {
+            it.attributes().replace("name", "One Dark Generated")
```

Comment:
Why One Dark Generated?

Reply:
![image](https://user-images.githubusercontent.com/15972415/83243843-19361e00-a164-11ea-83b7-cf5f77f2a23e.png)

I just wanted to make it look different from the other `One Dark` scheme.